pkg/storage/neonsan/api/api.go

```diff
@@ -296,11 +302,71 @@ func httpGet(confFile string, request interface{}, response Response) error {
 	return nil
 }
 
+func buildConfig(kubeconfig string) (*rest.Config, error) {
+	if kubeconfig != "" {
+		cfg, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+		if err != nil {
+			return nil, err
+		}
+		return cfg, nil
+	}
+
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
+func getActiveQfcenterFromK8s(cfgFile string) (string, error) {
+	config, err := toml.LoadFile(cfgFile)
+	if err != nil {
+		klog.Errorf("Config file error:%s\n", err.Error())
+		return "", err
+	}
+	kubeConfPath := config.GetDefault("k8s.conf", "").(string)
+	lockName := config.GetDefault("k8s.lock_name", "center").(string)
+	namespace := config.GetDefault("k8s.namespace", "default").(string)
+
+	kubeConfig, err := buildConfig(kubeConfPath)
+	if err != nil {
+		return "", err
+	}
+
+	clientset, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		return "", err
+	}
+
+	lease, err := clientset.CoordinationV1().Leases(namespace).Get(context.TODO(), lockName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	ip := *lease.Spec.HolderIdentity
+
+	if len(string(ip)) == 0 {
+		err := errors.New("No center found")
```

Comment:
An error with more info might be better

```suggestion
		err := fmt.Errorf("No leader center found in lease [%s/%s]", namespace, lockName)
```